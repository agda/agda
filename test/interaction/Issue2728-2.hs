{-# LANGUAGE RecordWildCards #-}

import Control.Exception

import RunAgda

file          = "Issue2728-2.agda"
firstPart s   = "F : {" ++ s ++ " : Set₁} → Set₁\nF {A} = A\n"
secondPart    = "G : Set₁\nG = F {A = Set}\n"
firstCode     = firstPart "_"
secondCode    = firstPart "_" ++ secondPart
thirdCode     = firstPart "A" ++ secondPart

main :: IO ()
main = runAgda [ "--no-libraries"
               , "--caching"
               ] $ \(AgdaCommands { .. }) -> do

  -- Discard the first prompt.
  echoUntilPrompt

  -- Load the file, and wait for Agda to complete.
  loadAndEcho file

  (do
    -- Edit the file and reload, twice.
    writeUTF8File file secondCode
    loadAndEcho file
    writeUTF8File file thirdCode
    loadAndEcho file)

    `finally`

    -- Restore the original contents of the file.
    writeUTF8File file firstCode

  return ()
