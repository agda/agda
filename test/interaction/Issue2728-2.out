(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue2728-2 (Issue2728-2.agda). " t)
(agda2-status-action "Checked")
(agda2-info-action "*All Done*" "" nil)
((last . 1) . (agda2-goals-action '()))
(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue2728-2 (Issue2728-2.agda). " t)
(agda2-info-action "*Error*" "Issue2728-2.agda:4,5-6: error: [WrongNamedArgument] Function does not accept argument {A = _} when checking that {A = Set} is a valid argument to a function of type {_ : Set₁} → Set₁" nil)
((last . 3) . (agda2-maybe-goto '("Issue2728-2.agda" . 46)))
(agda2-highlight-load-and-delete-action)
(agda2-status-action "")
(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue2728-2 (Issue2728-2.agda). " t)
(agda2-status-action "Checked")
(agda2-info-action "*All Done*" "" nil)
((last . 1) . (agda2-goals-action '()))
